MediaBrowser.MediaEncoding/Probing/FFProbeHelpers.cs

```diff
@@ -57,6 +57,31 @@ public static string GetDictionaryValue(IReadOnlyDictionary<string, string> tags
             return val;
         }
 
+        /// <summary>
+        /// Gets a string from an FFProbeResult tags dictionary, iterating through all keys,
+        /// stopping at the first key that returns a result that's neither null nor blank.
+        /// </summary>
+        /// <param name="tags">The tags.</param>
+        /// <param name="keys">The checked keys.</param>
+        /// <returns>System.String.</returns>
+        public static string GetFirstNonNullNorBlankDictionaryValue(IReadOnlyDictionary<string, string> tags, params string[] keys)
```

Comment:
The problem with this is that it allocates an array every time you call it

Reply:
Ah, so it's similar to Java where vararg creates an implicit copy, on top of the array declared at the callsite?

Reply:
Well, you haven't declared an array per se. You're just letting it convert the arguments to an array at runtime.

Reply:
Oh, then I meant something different.
Regardless, having a huge block of if checks is quite verbose, that's why I introduced that helper function. I could of course switch the `params` to just using an array, and declare the array with the dictionary keys statically in the file and not in the function.
That way, it creates less allocations when called multiple times. Would that work better?

Reply:
The options I see are

1. Create multiple methods with varying number of arguments
2. Create specialized methods for every parameter `GetPremiereDate(tags)`

Reply:
I created a [small benchmark](https://gist.github.com/Maxr1998/d5ea87ee42d8076311f835865f6da1c5) to test various alternatives. In all cases, the worst case of no results in the dictionary was used to test.

As expected, the array allocation create a huge performance hit, so that option should definitely be avoided. Declaring the array as class attribute, thus shared across allocations, performs quite well, but the overall memory usage of the program would increase, so maybe not the best option.
Using a hardcoded argument count performs best, so that's probably the route to take. I propose supporting 1, 2 and 3 parameters for the function. Specialized methods for each of the attributes (name, album artist, etc.) is imo a bit too verbose/specific.

Reply:
I implemented the suggested changes in 3e9570a6f55fff97724e5527a8fb39703b09d542. Should be good now.